src/webgpu/util/f32_interval.ts

```diff
@@ -1634,6 +1634,13 @@ export function mixPreciseInterval(x: number, y: number, z: number): F32Interval
   );
 }
 
+/** Calculate an acceptance interval of modf(x) */
+export function modfInterval(n: number): { fract: F32Interval; whole: F32Interval } {
+  const fract = correctlyRoundedInterval(n % 1.0);
```

Comment:
I doublechecked with ECMAScript to make sure this did what we want.

Floating point % in ECMAScript bottoms out in [Number::remainder](https://tc39.es/ecma262/multipage/ecmascript-data-types-and-values.html#sec-numeric-types-number-remainder)

E.g.    -5.5 % 1.0 yields   -0.5

That matches WGSL. So that's good.

Reply:
:+1: